doc/mgmt/SONiC_HAMD_RBAC_HLD.md

```diff
@@ -0,0 +1,1289 @@
+| operator (Network Operator) | Users must be allowed to execute only show commands. They must be granted read, write and execute permissions to all YANG xPATHs.Read-only permissions to all YANG xpaths. For certain objects, read-write access would be granted. For example, copying a file from the user home directory to a file server and vice-versa is permitted. Users must not be allowed to login into the Linux system shell. | operator, docker | sonic-cli |
+| secAdmin (security admin) | Users with secAdmin privileges must be allowed to access all security-related commands in the system.(configuration/show/debug/copy and etc.,) Read/Write/Execute permissions to the relevant YANG xpaths. For example, a secAdmin user must be allowed to access the commands related to aaa, authentication, radius server, ssh-server, TACACS server, CoPP, username and etc., And also the user must be allowed to access system-level information like login statistics, log information and etc., | secadmin, docker, redis | sonic-cli |
+| netAdmin (network admin) | Users with netAdmin privileges must be allowed to access the configuration commands/features that manage traffic flowing through the switch, such as routing protocols, route-map, bfd, qos, PFC, Session monitoring, SFLOW,  SFLOW, ACLs, interfaces, and etc. | netadmin, docker, redis | sonic-cli |
+
+
+### 1.3.2 Container
+
+ - No new containers are introduced for this feature. 
+ - Both sonic-mgmt-framework container (Rest) and sonic-telemetry container (gNMI) use translib library. So the required RBAC functionality would be implemented in the translib module (sonic-mgmt-common).
+	- When a REST query is received in mgmt-framework container, the URI is used to fetch the details from the RBAC Trie Tree and proceed further.
+	- When a gNMI GET/SET query is received in telemetry container, the URI is used to fetch the details from the RBAC Trie Tree and proceed further.
+
+
+### 1.3.3 SAI Overview
+
+No new or existing SAI services are required
+
+
+# 2 Functionality
+
+This feature is applicable to all north-bound interfaces like CLI/gNMI/RestAPI. So the RBAC feature must be implemented by a module that is common to all NBIs. Both sonic-mgmt-framework and telemetry containers use translib for writing/reading the configurations into/from the CONFIG DB. Hence, the RBAC feature would be implemented in the translib module.
+
+Both sonic-mgmt-framework (Rest) and sonic-telemetry(gNMI) use translib library. So the required RBAC functionality would be implemented in the translib module (sonic-mgmt-common).
+
+The following table captures the CLI commands/xPATHS and their access privileges to be granted to various roles.
+Note: If a command is applicable to all roles then the tag "access" must not be used.  The absence of the tag "access" means that the command is applicable to all.
+
+### Table 3: CLI commands and their access roles
```

Comment:
why do we need to have an explicit mention of admin role? admin role by definition always has read, write, execute access to every object, right ?

Reply:
Yes Arun, agreed ! 
We too didn't want to add the "admin" role in all the access tags.
We had made the required changes for "admin" role in a common API already.

Will double confirm the same and remove the "admin" role from all the access tags in the XMLs and update the HLD accordingly.